pydfirram/core/base.py

```diff
@@ -273,20 +327,24 @@ def __init__(self, operating_system: OperatingSystem, dump_file: Path):
         self.validate_dump_file(dump_file)
 
         self.os = operating_system
-        self.plugins: List[PluginEntry] = self.get_all_plugins()
+        self.plugins: list[PluginEntry] = self.get_all_plugins()
         self.dump_file = dump_file
-        self.context = None
+        self.context: Optional[Context] = None
         self.temp_data = None
-        self.tmp_plugin: PluginEntry = None
+        self.tmp_plugin: Optional[PluginEntry] = None
```

Comment:
Do we aim at supporting Python < 3.10 ? If not, we could use `|` operator instead of `Optional`

Reply:
`typing.Optional` is an alias of `X | None`, but is more explicit and appropriate in this context as defined in the documentation [here](https://docs.python.org/3.10/library/typing.html#typing.Optional)

> `Optional[X]` is equivalent to `X | None` (or `Union[X, None]`).
> [...]
> if an explicit value of None is allowed, the use of Optional is appropriate, whether the argument is optional or not
> ```python
> def foo(arg: Optional[int] = None) -> None:
> ...
> ```

And as for the minimum CPhython version, I have just checked the `pyproject.toml` information which indicates `python = "^3.10"` (which is, in my opinion, a good choice because of the addition of the syntaxic sugar `|` for `Union[]`  and various typing updates like `typing.Self`, deprecation of `typing.Dict`, `typing.List`, ...)